test/parse_test.js

```diff
@@ -18,28 +18,30 @@ const invalidAsyncpiJSON = fs.readFileSync(path.resolve(__dirname, './wrong/inva
-const invalidAsyncAPI = { asyncapi: '2.0.0', info: {} };
+const outputJsonWithRefs = '{"asyncapi":"2.0.0","info":{"title":"My API","version":"1.0.0"},"channels":{"mychannel":{"publish":{"traits":[{"externalDocs":{"url":"https://company.com/docs"}}],"externalDocs":{"x-extension":true,"url":"https://irrelevant.com"},"message":{"traits":[{"x-some-extension":"some extension"}],"payload":{"type":"object","properties":{"name":{"type":"string"},"test":null}}}}},"oneOfMessageChannel":{"publish":{"message":{"oneOf":[{"traits":[{"x-some-extension":"some extension"}],"payload":{"type":"object","properties":{"name":{"type":"string"},"test":null}}}]}}}},"components":{"messages":{"testMessage":{"traits":[{"x-some-extension":"some extension"}],"payload":{"type":"object","properties":{"name":{"type":"string"},"test":null}}}},"schemas":{"testSchema":{"type":"object","properties":{"name":{"type":"string"},"test":null}}},"messageTraits":{"extension":{"x-some-extension":"some extension"}},"operationTraits":{"docs":{"externalDocs":{"url":"https://company.com/docs"}}}}}';
+const invalidAsyncAPI = '{"asyncapi":"2.0.0","info":{}}';
 
 const eolLength = EOL.length;
 
-const checkErrorTypeAndMessage = async (fn, type, message) => {
-  try {
-    await fn();
-    throw Error('should not be reachable');
-  } catch (e) {
-    expect(e instanceof ParserError).to.equal(true);
-    expect(e).to.have.own.property('type', type);
-    expect(e).to.have.own.property('message', message);
-  }
-};
+const checkErrorWrapper = async (fn, validationObject) => {
+  const { type, message, title, refs, detail, location, validationErrors, parsedJSON } = validationObject;
 
-const checkErrorParsedJSON = async (fn, parsedJSON) => {
   try {
     await fn();
-    throw Error('should not be reachable');
+    throw Error('This error should not be reachable. If you reached it, it means the function did not throw a proper error and executed successfully.');
   } catch (e) {
-    expect(JSON.stringify(e.parsedJSON)).to.equal(parsedJSON);
+    const isProperError = e instanceof ParserError;
+    if (!isProperError) console.log(e);
+
+    if (isProperError) expect(e instanceof ParserError).to.equal(true);
```

Comment:
I added this part to make sure that when something really goes wrong and the error from line 31 is reached a clear and meaningfull error is actually visible to the user. Now it wasn't